### PR TITLE
Fix COCO configuration

### PR DESCRIPTION
**Pull request by chienyiwang**
I tried the training and testing on COCO dataset and fixed some minor bugs in the configuration.

### CODE REVIEW

**Comment by ddonatien** [comment]
Hi @d-li14 I hope you have solved your problem by now, but just for the record you can find other sources of the file as stated in issue #205 and #123

**Comment by d-li14** [comment]
Thanks @ddonatien , but as far as I'm concerned, as stated in #123, just replacing instances_trainval_35k.json with valminusminival2014.json seems not reasonable.

**Comment by foralliance** [comment]
@d-li14 
@ddonatien 

replacing **instances_trainval_35k.json** with **valminusminival2014.json** is **wrong**.

Theoretically, the **instances_trainval_35k.json** should contain 118287 images. If you do this substitution, the **instances_trainval_35k.json** only contain 35185 images

**Comment by SalahAdDin** [comment]
It does not work to me then, the script is asking fot the json file, but it is a **zip** that cannot be unzipped.

**Comment by Damon2019** [comment]
I tried the training on COCO dataset ,can anyone tell me how to modify the coco_labels.txt?

**Comment by 814247846** [comment]
I think  instances_trainval_35k.json is the COCO2017 instances_train2017.json.Beacuse it's have 118287 images.So trainval_35k is just the COCO2017train.

**Comment by 814247846** [comment]
I think :1.trainval35k is COCO 2017 train,because it have 118287 images.
2.So the instances_trainval35k.json is instances_train2017.json

**Comment by SalahAdDin** [comment]
@814247846 Did you get work with that `json` file?

**Comment by 814247846** [comment]
> @814247846 Did you get work with that json file?

yes.But you also have to change:
img = cv2.imread(osp.join(self.root, path))---->img = cv2.imread(path)
I think it's possible that the author made a mistake.

**Comment by SalahAdDin** [comment]
@814247846 all PR are welcomed.

**Comment by MiChuan** [comment]
The data sets include 2014 (2014 Train images, 2014 Val images, 2014 Test images), 2015 (2015 Test images) and 2017 (2017 Train images, 2017 Val images, 2017 Test images). Among them, training set and validation set There are tags (2014 Train/Val annotations, 2017 Train/Val annotations), while the test set has no tags.

In the 2014 data set, the training set is 82,783, the validation set is 40504, and the test set is 40775. In addition, the validation set is divided into two parts, miniVal has 5000, and the remaining 35504 images and training set are called Trainval35k (Trainval35k = = train2014 + val2014 - minival2014.). Trainval35k is usually used as the training set in the paper.

There are 81,434 images in the 2015 test set.

In the 2017 data set, the training set is 118287, the verification is 5000, and the test is 40670. Among them, **the training set is Trainval35k (82783 + 40504 - 5000 = 118287)**, and the verification set is miniVal. That is, **train2017 == trainval35k == train2014 + val2014 - minival2014 == train2014 + val2014 - val2017**.
----------------------------------------------------------------------------------------------
Original link: https://blog.csdn.net/qq_37643960/article/details/103466904
